src/components/client/campaign-expenses/Form.tsx

```diff
@@ -0,0 +1,269 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+import { useRouter } from 'next/router'
+import { useTranslation } from 'next-i18next'
+import { AxiosError, AxiosResponse } from 'axios'
+import * as yup from 'yup'
+import { FormikHelpers } from 'formik'
+
+import { Box, Button, Grid, Tooltip, Typography } from '@mui/material'
+
+import { routes } from 'common/routes'
+import { Currency } from 'gql/currency'
+import { AlertStore } from 'stores/AlertStore'
+import { endpoints } from 'service/apiEndpoints'
+import LinkButton from 'components/common/LinkButton'
+import { useViewExpense } from 'common/hooks/expenses'
+import GenericForm from 'components/common/form/GenericForm'
+import SubmitButton from 'components/common/form/SubmitButton'
+import CurrencySelect from 'components/common/currency/CurrencySelect'
+import FormTextField from 'components/common/form/FormTextField'
+import { Checkbox } from '@mui/material'
+import { useCreateExpense, useEditExpense } from 'service/expense'
+
+import { ApiErrors, isAxiosError, matchValidator } from 'service/apiErrors'
+import { ExpenseInput, ExpenseResponse, ExpenseStatus, ExpenseType } from 'gql/expenses'
+import FileUpload from 'components/common/file-upload/FileUpload'
+
+import ExpenseTypeSelect from 'components/admin/expenses/ExpenseTypeSelect'
+import { useViewCampaign } from 'common/hooks/campaigns'
+import { UploadExpenseFile, ExpenseFile } from 'gql/expenses'
+import { useUploadExpenseFiles } from 'service/expense'
+import FileList from 'components/client/campaign-expenses/grid/FileList'
+import FormDatePicker from 'components/common/form/FormDatePicker'
+import { toMoney, fromMoney } from 'common/util/money'
+import { useCampaignExpenseFiles } from 'common/hooks/expenses'
+import { downloadCampaignExpenseFile, deleteExpenseFile } from 'service/expense'
+import { useSession } from 'next-auth/react'
+import DeleteForeverIcon from '@mui/icons-material/DeleteForever'
+import { useViewPersonByKeylockId } from 'common/hooks/person'
+
+const validTypes = Object.keys(ExpenseType)
+const validStatuses = Object.keys(ExpenseStatus)
+const validCurrencies = Object.keys(Currency)
+
+export default function Form() {
+  const queryClient = useQueryClient()
+  const router = useRouter()
+  const slug = router.query.slug as string
+  const { t } = useTranslation('expenses')
+  const { data: campaignResponse } = useViewCampaign(slug)
+  let id = router.query.id as string
+  const [files, setFiles] = useState<File[]>([])
+  const { data: expenseFiles } = useCampaignExpenseFiles(id)
+  const { data: session } = useSession()
+
+  const isAdmin = session?.user?.realm_access?.roles?.includes('podkrepi-admin')
+
+  const { data: person } = useViewPersonByKeylockId(session?.user?.sub as string)
+
+  const fileUploadMutation = useMutation<
+    AxiosResponse<ExpenseFile[]>,
+    AxiosError<ApiErrors>,
+    UploadExpenseFile
+  >({
+    mutationFn: useUploadExpenseFiles(),
+  })
```

Comment:
the file upload might need an onError handler, because when I was trying locally it failed due to the missing bucket, but didn't show an error

Reply:
I added it. But don't think it helps.
Catching those kind of error seems to be tricky.